src/lib/client/Client.ts

```diff
@@ -312,3 +326,66 @@ export class Client extends BaseClient {
+	[ClientEvents.ChannelPinsUpdate](channel: TextBasedChannel, updatedAt: Date | null): void;
+	[ClientEvents.GuildCreate](guild: Guild): void;
+	[ClientEvents.GuildUpdate](guild: Guild, previousGuild?: Guild): void;
+	[ClientEvents.GuildDelete](guild: Guild): void;
+	[ClientEvents.GuildBanAdd](ban: Ban): void;
+	[ClientEvents.GuildBanRemove](ban: Ban): void;
+	[ClientEvents.GuildEmojiCreate](emoji: GuildEmoji, guild: Guild): void;
+	[ClientEvents.GuildEmojiDelete](emoji: GuildEmoji, guild: Guild): void;
+	[ClientEvents.GuildEmojisUpdate](emoji: GuildEmoji, previousEmoji: GuildEmoji, guild: Guild): void;
+	[ClientEvents.GuildIntegrationsUpdate](guild: Guild, previousGuild: Guild): void;
+	[ClientEvents.GuildMemberAdd](guildMember: GuildMember): void;
+	[ClientEvents.GuildMemberRemove](guildMember: GuildMember): void;
+	[ClientEvents.GuildMemberUpdate](guildMember: GuildMember, previousGuildMember: GuildMember): void;
+	[ClientEvents.GuildMembersChunk](members: GuildMember[], guild: Guild, extraData: { chunkCount?: number, chunkIndex?: number, nonce?: number }): void;
+	[ClientEvents.GuildRoleCreate](role: Role): void;
+	[ClientEvents.GuildRoleDelete](role: Role): void;
+	[ClientEvents.GuildRoleUpdate](role: Role, previousRole?: Role): void;
+	[ClientEvents.InviteCreate](invite: Invite): void;
+	[ClientEvents.InviteDelete](invite: Invite): void;
+	[ClientEvents.MessageCreate](message: Message): void;
+	[ClientEvents.MessageDelete](message: Message, previousMessage?: Message): void;
+	[ClientEvents.MessageUpdate](message: Message, previousMessage: Message): void;
+	[ClientEvents.MessageDeleteBulk](messages: ({ id: string } | Message)[], channel: TextBasedChannel): void;
+	[ClientEvents.MessageReactionAdd](reaction: MessageReaction, user: User): void;
+	[ClientEvents.MessageReactionRemove](reaction: MessageReaction, user: User): void;
+	[ClientEvents.MessageReactionRemoveAll](message: Message, reactions: Cache<string, MessageReaction>): void;
+	[ClientEvents.MessageReactionRemoveEmoji](reaction: MessageReaction): void;
+	[ClientEvents.PresenceUpdate](presence: Presence, previousPresence: Presence | null): void;
+	[ClientEvents.TypingStart](channel: GuildBasedChannel | DMChannel, user: User): void;
+	[ClientEvents.UserUpdate](user: User): void;
+	[ClientEvents.VoiceStateUpdate](voiceState: VoiceState): void;
+	[ClientEvents.VoiceServerUpdate](voiceServerData: VoiceServerUpdateDispatch): void;
+	[ClientEvents.WebhooksUpdate](channel: TextChannel | NewsChannel, previousChannel: TextChannel | NewsChannel): void;
+	[ClientEvents.Debug](message: string): void;
+	[ClientEvents.RESTDebug](message: string): void;
+	[ClientEvents.WSDebug](message: string): void;
+	[ClientEvents.Error](error: Error): void;
+	[ClientEvents.ShardOnline](): void;
+}
+
+export declare interface Client {
+	addListener<K extends keyof ClientEventsMap>(event: K, listener: ClientEventsMap[K]): this;
+	eventNames(): (keyof ClientEventsMap)[];
+	getMaxListeners(): number;
```

Comment:
This line and line 390 are redundant as they don't depend on events

Reply:
resolved in 1fbf517